src/containers/Accounts/AccountsRouter.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { useParams } from 'react-router'
+import AMMAccounts from 'containers/Accounts/AMM/AMMAccounts/index'
+import { connect } from 'react-redux'
+import SocketContext from '../shared/SocketContext'
+import { getAccountInfo } from '../../rippled/lib/rippled'
+import NoMatch from '../NoMatch'
+import Accounts from './index'
+import { ERROR_MESSAGES } from './Errors'
+import Loader from '../shared/components/Loader'
+import { ACCOUNT_FLAGS } from '../../rippled/lib/utils'
+
+const getErrorMessage = (error: any) =>
+  ERROR_MESSAGES[error] || ERROR_MESSAGES.default
+
+function renderError(error: any) {
+  const message = getErrorMessage(error.code)
+  return (
+    <div className="accounts-page">
+      <NoMatch title={message.title} hints={message.hints} />
+    </div>
+  )
+}
+
+const AccountsRouter = () => {
+  const { id: accountId } = useParams<{ id: string }>()
+  const rippledSocket = useContext(SocketContext)
+  const [error, setError] = useState<any>()
+  const [comp, setComp] = useState<any>()
+  const flags: any = Object.entries(ACCOUNT_FLAGS).reduce(
+    (all, [key, value]) => ({ ...all, [value]: key }),
+    {},
+  )
+
+  useEffect(() => {
```

Comment:
The way this is setup `getAccountInfo` is called twice.  Once here and then in the eventual component.

Reply:
That is true.